In [1]:
import pandas as pd

In [2]:
head_df = pd.read_csv('E:\Python\Machine Learning\datasets\headsize.csv')

In [3]:
head_df.columns =['Gender', 'Age Range', 'Head_Size_cm', 'Brain_Weight_grams']

In [4]:
head_df.shape

(237, 4)

In [5]:
head_df.describe()

Gender   Age Range  Head_Size_cm  Brain_Weight_grams
count  237.000000  237.000000    237.000000          237.000000
mean     1.434599    1.535865   3633.991561         1282.873418
std      0.496753    0.499768    365.261422          120.340446
min      1.000000    1.000000   2720.000000          955.000000
25%      1.000000    1.000000   3389.000000         1207.000000
50%      1.000000    2.000000   3614.000000         1280.000000
75%      2.000000    2.000000   3876.000000         1350.000000
max      2.000000    2.000000   4747.000000         1635.000000

#### describe gives information about average 3633.99 and 1282.87 for headsize and brain weight respectiverly. However standard deviation is 365.26 ans 120.34 for headsize and brain weight.

In [ ]:
import seaborn as sns

In [ ]:
sns.distplot(head_df['Head_Size_cm'])

In [ ]:
sns.distplot(head_df['Brain_Weight_grams'])

In [ ]:
from statsmodels.stats import weightstats
import scipy.stats as stats
import numpy as np

# Implement One Sample Z-test

In [ ]:
sample_z_index = np.random.randint(0,237,47)

In [ ]:
random_z_sample = head_df.iloc[sample_z_index]

In [ ]:
random_z_sample.describe()

### Observation : 
    mean of Head_Size_cm for randomly selected 47 sample record is 3667.63, where as mean of complete data is 3633.99 
    
    here, we may raise the question as whether mean of given sample and mean of total population has changed? to resolve this query we may implement hypothesis testing. The sample contains 47 records hence z-test can be implemented.
    

#### NULL : Mean of sample is same as mean of population
#### Alternate: Mean of sample and mean of population is different

In [ ]:
z_statistic_val, prob = weightstats.ztest(random_z_sample['Head_Size_cm'], value=3633.99 )

In [ ]:
NULL='Mean of sample is same as mean of population'
alternate= 'Mean of sample and mean of population is different'
if prob < 0.05:
    print("Reject NULL:", NULL )
    print("Accept alternate",alternate)
else:
    print("Accept NULL:-", NULL)
    print("Reject alternate:-", alternate)


#### Hence, it is prooved that mean of sample and mean of population is the same.

# Implement  One SampleT-test

In [ ]:

sample_index = np.random.randint(0,237,7)

In [ ]:
sample_index

In [ ]:
random_sample = head_df.iloc[sample_index]

In [ ]:
random_sample

In [ ]:
random_sample.describe()